nilearn/decomposition/multi_pca.py

```diff
@@ -211,7 +230,7 @@ def fit(self, imgs, y=None, confounds=None):
             raise ValueError('Need one or more Niimg-like objects as input, '
                              'an empty list was given.')
         if confounds is None:
-            confounds = itertools.repeat(None, len(imgs))
+            confounds = [None] * len(imgs)  # itertools.repeat(None, len(imgs))
```

Comment:
Yes, for some reason `confounds` end up being `repeat(None, 0)` on the following line. I don't know how this come from, I just figured it would just be pragmatic to change it.